sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala

```diff
@@ -281,89 +281,108 @@ package object dsl {
       def attr: UnresolvedAttribute = analysis.UnresolvedAttribute(s)
 
       /** Creates a new AttributeReference of type boolean */
-      def boolean: AttributeReference = AttributeReference(s, BooleanType, nullable = true)()
+      def boolean: AttributeReference =
+        AttributeReference(attr.nameParts.last, BooleanType)(qualifier = attr.nameParts.init)
```

Comment:
can add a new method to create attribute to deduplicate code? The only difference is the data type.